.github/workflows/publish-artifacts.yml

```diff
@@ -0,0 +1,173 @@
+name: Publish artifacts to ORAS
+
+on:
+  push:
+    branches:
+    - main
+
+env:
+  RUST_TOOLCHAIN: 1.80.0
```

Comment:
I like the idea of pining the rust version (IIRC most of the other builds are using stable and image-rs is using 1.76?). This is probably outside the scope of this PR, but I wondered if having the versions in a shared place (e.g. versions.yaml) would be helpful for sync these up. 

Also was there are particular reason for picking 1.80.0 over 1.81.0 as the latest stable?
I'll note that in kata-containers we are currently using rust 1.76.0 as Red Hat couldn't bump to a more recent versions for their downstream work. This isn't necessarily a reason to downgrade, I just wanted to share the reasons behind kata being back-level.

Reply:
yeah, it's a bit unfortunate that we don't have this centralized somewhere. we could also consider a composite action that specifies the rust version in a single place at least for the CI actions. It's not that critical to be 100% consistent, since rust doesn't have a runtime, but it's annoying when something breaks.

tha latest stable 1.81.0 is broken for some dependency that we use I think.

Reply:
Cool - that sounds like a good reason to not pick 1.81.0!